src/components/Warehouse/WarehouseDetailsContainer.js

```diff
@@ -0,0 +1,85 @@
+import editIcon from "../../assets/icons/edit-white-24px.svg";
+import "./WarehouseDetailsContainer.scss";
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import BackIcon from "../../assets/icons/arrow_back-24px.svg";
+import { useParams } from "react-router-dom";
+import { useNavigate } from "react-router-dom";
+
+const WarehouseDetailsContainer = () => {
+  const navigate = useNavigate();
+  const [warehouseData, setWarehouseData] = useState([]);
+  const { id } = useParams();
+
+  // This useEffect fetches the Warehouse data when the component mounts
+  useEffect(() => {
+    const fetchWarehouse = async () => {
+      try {
+        const resp = await axios.get(
+          `http://localhost:8080/api/warehouses/${id}`
+        );
+        setWarehouseData(resp.data);
+      } catch (error) {
+        console.error("Error fetching Warehouse data:", error);
+      }
+    };
+    fetchWarehouse();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <>
+      <section className="detail-card-container">
+        <div className="detail-card-container__header-box">
+          <img
+            onClick={() => {
+              navigate(`/warehouses`);
+            }}
+            src={BackIcon}
+            alt="Edit Icon"
+            className="form-cta-details__back-icon"
+          />
+          <h1 className="detail-card-container__heading">
+            {warehouseData.warehouse_name}
+          </h1>
+          <button className="form-cta-details">
+            <img
+              src={editIcon}
+              alt="Edit Icon"
+              className="form-cta-details__icon"
+            />
+            <p className="form-cta-details__text">Edit</p>
```

Comment:
Semantically this <p> tag is not valid because you have it in a button. For styling within button `span` is more appropriate